agreementbot/secrets/vault/vault.go

```diff
@@ -28,19 +29,32 @@ func (vs *AgbotVaultSecrets) String() string {
 	return fmt.Sprintf("Token: %v", vs.token)
 }
 
-// This utility will be available to any users within the org.
-func (vs *AgbotVaultSecrets) ListOrgSecret(user, password, org, name string) (map[string]string, error) {
+// Available to all users within the org
+func (vs *AgbotVaultSecrets) ListOrgUserSecret(user, password, org, name string) (map[string]string, error) {
+	glog.V(3).Infof(vaultPluginLogString(fmt.Sprintf("list secret %v for org %v user %v", name, org, user)))
+
+	_, userId := cutil.SplitOrgSpecUrl(user)
+	url := fmt.Sprintf("%s/v1/openhorizon/%s/user/%s/%s", vs.cfg.GetAgbotVaultURL(), org, userId, name)
+	return vs.ListSecret(user, password, org, name, url)
+}
 
+// Available to only org admin users
+func (vs *AgbotVaultSecrets) ListOrgSecret(user, password, org, name string) (map[string]string, error) {
 	glog.V(3).Infof(vaultPluginLogString(fmt.Sprintf("list secret %v for org %v", name, org)))
 
+	url := fmt.Sprintf("%s/v1/openhorizon/%s/%s", vs.cfg.GetAgbotVaultURL(), org, name)
+	return vs.ListSecret(user, password, org, name, url)
+}
+
+// Get the secret at a specified path within the vault
+func (vs *AgbotVaultSecrets) ListSecret(user, password, org, name, url string) (map[string]string, error) {
```

Comment:
This function name should really be the lowercase name listSecret so that it stays private to the package.